tests/utils/test_environment.py

```diff
@@ -326,3 +327,31 @@ def make_conda_env(pip_deps):
 
     with pytest.raises(TypeError, match=r"Expected .+, but got `int`"):
         _process_conda_env(0)
+
+
+def test_duplicate_pip_requirements():
+    packages = [
+        "numpy==1.21.0",  # specific version
+        "pandas>=1.3.0",  # minimum version
+        "scikit-learn",  # any version
+        "matplotlib<=3.4.2",  # maximum version
+        "seaborn",  # any version
+        "package!=1.2.3",  # any version but this one
+        "package~=1.2.3",  # compatible version (i.e., >= 1.2.3, == 1.2.*)
+        "package>1.2.3",  # version greater than
+        "package<1.2.3",  # version less than
+        "package[extra]==1.2.3",  # specific version with extras
+        "fastapi ===0.21.0",  # specific version (PEP 440: version with spaces around ===)
+        "fastapi @ https://my.package.repo/fastapi-0.21.0-py3-none-any.whl",  # private repo
+        "-e git+https://github.com/mlflow/mlflow@master",  # editable mode, direct from a git repo
+        "-r requirements.txt",  # from a requirements file
+        "pytest-cov; python_version < '3.8'",  # conditional requirements
+        "fastapi @ file:///local/path/to/numpy-0.22.0-py3-none-any.whl",  # wheel from a local file
+        "-c constraints.txt",  # constraint file
+        "--no-binary :all:",  # no binary packages, source code only
+        "--prefer-binary",  # prefer binary packages over source code
+        "numpy<1.24.0",  # maximum version
+        "numpy",  # any version
+    ]
+    evaluation = _find_duplicate_requirements(packages)
+    assert sorted(evaluation) == ["numpy", "package"]
```

Comment:
Can we test that a warning is emitted when we call `mlflow.transformers.log_model()`? That way, we know `_find_duplicate_requirements` is called properly by `mlflow.transformers.log_model()` :)

Reply:
(Feel free to mock `logger.warning` :D )